src/openms_gui/source/VISUAL/LayerData.cpp

```diff
@@ -134,6 +142,35 @@ namespace OpenMS
     }
   }
 
+  bool LayerData::annotate(const vector<PeptideIdentification>& identifications,
+    const vector<ProteinIdentification>& protein_identifications)
+  {
+    IDMapper mapper;
+    if (this->type == LayerData::DT_PEAK)
```

Comment:
can you add an issue or TODO that IDMapper should support mapping by native ID if available? Would be much safer

Reply:
good point
#4965